packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -66,15 +67,52 @@ const ProductTile = (props) => {
         ...rest
     } = props
 
-    const {currency, image, price, productId, hitType} = product
+    const {image, productId, hitType} = product
+
+    // NOTE: swatches will implement later to set variant accordingly,
+    // On first load, get the variant that is the represent product
+    // this is for variant product, standard/set/bundles does not have variants
+    // Also, product tile can be used in RecommendedProducts where it calls getProducts which does not have representedProduct
+    // in that case we use the first variant that has price book to set up discount price
+    // Not all variants has set in a priceBook, meaning not having tieredPrices.
+    const variant = product?.variants?.find(
+        (i) => i?.productId === product?.representedProduct?.id || !!i?.tieredPrices
+    )
+
+    // prioritize variant promotionalPrice over standard price
+    let currentPrice = variant
+        ? getDisplayPrice(variant)?.discountPrice || variant?.price
+        : product?.price
 
     // ProductTile is used by two components, RecommendedProducts and ProductList.
     // RecommendedProducts provides a localized product name as `name` and non-localized product
     // name as `productName`. ProductList provides a localized name as `productName` and does not
     // use the `name` property.
     const localizedProductName = product.name ?? product.productName
 
-    const {currency: activeCurrency} = useCurrency()
+    // check for both data returned from getProducts and productSearch
+    const isProductASet = hitType === 'set' || !!product?.type?.set
+    const isProductABundle = hitType === 'bundle' || !!product?.type?.bundle
+    const isProductAStandard = hitType === 'product' || !!product?.type?.item
+    let tieredPrices
+    if (variant) {
+        tieredPrices = variant?.tieredPrices
+    } else if (isProductABundle || isProductAStandard) {
+        tieredPrices = product?.tieredPrices
```

Comment:
Is this required if the result is the same as the else?

Reply:
What do you mean the same here? 🤔